pytorch_lightning/trainer/training_tricks.py

```diff
@@ -169,7 +169,7 @@ def scale_batch_size(self,
                 f' If this is not the intended behavior, please remove either one.'
             )
 
-        if hasattr(model.train_dataloader(), 'patch_loader_code'):
+        if hasattr(model.train_dataloader, 'patch_loader_code'):
```

Comment:
why doesn't it allow train_dataloader passed to `.fit`??

Reply:
don't know right now, maybe @SkafteNicki wrote it.

Reply:
actually, now that I think about it, @rohitgr7 it is because when train dataloader is defined from the outside, we can't set the batch size, there is no such attribute we can update. In fact, if you try to do so on an instance of DataLoader, it will complain.

Reply:
yeah, got it :+1